### ISSUE TITLE
Consider adding support for re-exports to `no-types-missing-file-annotation`

### ISSUE THREAD

**Issue by pascalduez** [gajus/eslint-plugin-flowtype#227]
Hi,

the `no-types-missing-file-annotation` does not support re-exports declarations, although this is quite an handy pattern. I've been caught a couple of times with this (forgetting the flow file declaration in an index), and such warning would have been of great help.

```js
`Button.js`

// @flow

export type Props = { ... }
```

```js
`index.js`

export type { Props } from './Button.js';
```
--> Should warn for missing flow file declaration.

**Comment by pascalduez** [comment]
`eslint-plugin-flowtype@2.33.0` 

;)

**Comment by gajus** [comment]
Thank you, @pascalduez
